### ISSUE TITLE
Gridmap static memory leak

### ISSUE THREAD

**Issue by me2beats** [godotengine/godot#57746]
### Godot version

3.4.2

### System information

Windows, Linux

### Issue description

static memory grows unexpectedly when setting gridmap cell to -1 (which means removing the cell)

### Steps to reproduce

1. create any meshlib (I made it from a scene with a single CubeMesh, without any material/collision etc)
2. create a scene with gridmap with this script:
```swift
extends GridMap

func _physics_process(delta):
    set_cell_item(0,0,0,0)
    set_cell_item(0,0,0,-1)
```
3. run the scene

As you can see in debugger, static memory slowly grows.
Also it's not cleared when using clear()/queue_free()

**Comment by me2beats** [comment]
Godot 4 also has this problem

**Comment by Kyrbulys** [comment]
![8x8_Oct8](https://user-images.githubusercontent.com/99164888/152774834-40d9e8b4-6021-4407-8e09-9bf798ed5186.png)
![16x16_Oct8](https://user-images.githubusercontent.com/99164888/152774862-ac74f58a-2994-460d-9641-169ef2324ba5.png)
![16x16_Oct16](https://user-images.githubusercontent.com/99164888/152774870-c99ccc2d-e54c-4852-be45-9264abcbb501.png)
![64x64_Oct8](https://user-images.githubusercontent.com/99164888/152774895-f4b8809e-9404-44a6-94c4-e9923d69016c.png)
![64x64_Oct64](https://user-images.githubusercontent.com/99164888/152774906-ada7d51f-7c71-4cff-8a25-5b333dceff10.png)
![32x32_Oct8_Bullet](https://user-images.githubusercontent.com/99164888/152774928-0553182d-6e74-4bc8-862a-fd7dd0e69358.png)

**Check project** run on Godot Physics 3.3.3 - 3.4.2.
[GridMapIssue.zip](https://github.com/godotengine/godot/files/8014607/GridMapIssue.zip)



Acknowledged. Tested it on versions 3.3.3 - 3.4.2. Windows x64. 
Here's what memory leak looks like during sample events in game with world generation.
Doesn't help clear() -> queue_free() of GridMap, and scene change either. The memory stays clogged.
Checked my project on different conditions, least leaks in 8x8 chunk and Octant 8. But throughout they appear. unfortunately the size of such a small chunk is not much of any game will fit.
Less leaks on BULLET, but the speed of 12-20ms in 32x32 in 1 frame will not allow to use it. 
The higher the chunk size, the more memory leakage.
The nature of leaks at chunks higher than 8x8 is always the same - the next chunk and memory addition size between chunks is always about 2-2.5 times larger than the previous one.

Very unpleasant bug!

**Comment by lawnjelly** [comment]
Already fixed in 3.x by #57561.

There's some slight juggling of static memory still now but no major leaks. The fix is pending in 4.0 as part of #57630.

**Comment by Kyrbulys** [comment]
> Already fixed in 3.x by #57561. (Notice the similarity with #57555)
> 
> Not absolutely sure if the fix was in by 3.4.3 RC 1 but should be in next 3.4.3, and next 3.5 beta.
> 
> There's some slight juggling of static memory still now but the major leak from BVH is gone. The fix is pending in 4.0 as part of #57630.

Checked on 3.4.3 RC1 - still same

**Comment by lawnjelly** [comment]
Yeah I just checked, the fix must have just missed that release, it was only merged day before I think or so. Will be in the next one. :+1:
